lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,483 @@
+            lightning_thresholds=set_lightning_thresholds))
+        msg = ("""<NowcastLightning: radius={radius}, debug={debug},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+With:
+<ApplyPrecip:
+ precipitation mapping:
+   upper:  precip probability {precu} => max lightning prob {lprecu}
+   middle: precip probability {precm} => max lightning prob {lprecm}
+   lower:  precip probability {precl} => max lightning prob {lprecl}
+
+   heavy:  prob(precip>7mm/hr)  {pphvy} => min lightning prob {lprobl}
+   intense:prob(precip>35mm/hr) {ppint} => min lightning prob {lprobu}
+>
+<ApplyIce:
+ VII (ice) mapping (kg/m2):
+   upper:  VII {viiu} => max lightning prob {lviiu}
+   middle: VII {viim} => max lightning prob {lviim}
+   lower:  VII {viil} => max lightning prob {lviil}
+>
+>""".format(radius=10000., debug=False,
+            lthru=set_lightning_thresholds[0], lthrl=0.,
+            lprobu=1., lprobl=0.25,
+            precu=0.1, precm=0.05, precl=0.0,
+            lprecu=1., lprecm=0.2, lprecl=0.0067,
+            pphvy=0.4, ppint=0.2,
+            viiu=2.0, viim=1.0,
+            viil=0.5,
+            lviiu=0.9, lviim=0.5,
+            lviil=0.1)
+              )
+        self.assertEqual(result, msg)
+
+
+class Test__process_haloes(IrisTest):
+
+    """Test the _process_haloes method."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point."""
+        self.cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube())
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin._process_haloes(self.cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_input(self):
+        """Test that the method does not modify the input cube."""
+        plugin = Plugin()
+        incube = self.cube.copy()
+        plugin._process_haloes(incube)
+        self.assertArrayAlmostEqual(incube.data, self.cube.data)
+
+    def test_data(self):
+        """Test that the method returns the expected data"""
+        plugin = Plugin(3000.)
+        expected = self.cube.data.copy()
+        expected[0, 0, 6, :] = [1., 1., 1., 1., 1., 1., 11./12., 0.875,
+                                11./12., 1., 1., 1., 1., 1., 1., 1.]
+        expected[0, 0, 7, :] = [1., 1., 1., 1., 1., 1., 0.875, 5./6., 0.875,
+                                1., 1., 1., 1., 1., 1., 1.]
+        expected[0, 0, 8, :] = [1., 1., 1., 1., 1., 1., 11./12., 0.875,
+                                11./12., 1., 1., 1., 1., 1., 1., 1.]
+        result = plugin._process_haloes(self.cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+
+class Test__update_meta(IrisTest):
+
+    """Test the _update_meta method."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point."""
```

Comment:
Again, need to know what coordinates & attributes are on the input cube to understand the tests.